### PR TITLE
Merged armv8 arch into basic sweb

### PR DESCRIPTION
**Pull request by TheAngusMcFire**
Everything works on my side with a fresh clone of the repo.

### CODE REVIEW

File: arch/armv8/rpi3/CMakeLists.compiler [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+
+# this one is important
+SET(CMAKE_SYSTEM_NAME Generic)
+SET(CMAKE_TRY_COMPILE_TARGET_TYPE STATIC_LIBRARY)
+
+# which compilers to use for C and C++
+
+find_program(CMAKE_ASM_COMPILER aarch64-linux-gnu-gcc)
+find_program(CMAKE_C_COMPILER aarch64-linux-gnu-gcc)
+find_program(CMAKE_CXX_COMPILER aarch64-linux-gnu-g++)
```
Comment (Woazboat) [inline arch/armv8/rpi3/CMakeLists.compiler]: Using a compiler that targets linux could be problematic. `aarch64-none-elf-gcc` would probably be better.

File: arch/armv8/common/include/ArchThreads.h [hunk 2]
```diff
@@ -0,0 +1,140 @@
...
+ * @param info where the ArchThreadRegisters is saved
+ * @param start_function instruction pointer is set so start function
+ * @param user_stack pointer to the userstack
+ * @param kernel_stack pointer to the kernel stack
+ */
+  static void createUserRegisters(ArchThreadRegisters *&info, void* start_function, void* user_stack, void* kernel_stack);
+
+/**
+ *
+ * on x86: invokes int65, whose handler facilitates a task switch
```
Comment (Woazboat) [inline arch/armv8/common/include/ArchThreads.h]: x86 specific comment

File: arch/armv8/common/include/ArchThreads.h [hunk 3]
```diff
@@ -0,0 +1,140 @@
...
+ *
+ */
+  static void yield();
+
+/**
+ * sets a threads CR3 register to the given page dir / etc. defining its address space
```
Comment (Woazboat) [inline arch/armv8/common/include/ArchThreads.h]: x86 specific comment

**Comment by dgruss** [comment]
On 01.10.2018 22:27, Woazboat wrote:
> Using a compiler that targets linux could be problematic.
> |aarch64-none-elf-gcc| would probably be better.

Yes, that must be fixed.

**Comment by Woazboat** [comment]
There also seems to be some code duplication. Maybe folder structure could be reorganized in the future to improve that situation.

e.g.
arch/armv8/common/source/assert.cpp
arch/armv8/common/source/IDEDriver.cpp
arch/armv8/common/include/atkbd.h
arch/armv8/common/source/atkbd.cpp

**Pull request merged**